project/AutoTLSCross.scala

```diff
@@ -0,0 +1,141 @@
+import sbt._
+import sbt.Keys._
+import sbt.internal.CommandStrings
+import sbt.internal.util.complete.DefaultParsers._
+import sbt.internal.util.complete.{DefaultParsers, Parser}
+import sbt.internal.CommandStrings._
+import sbt.Def.{ ScopedKey, Setting }
+
+object AutoTLSCross {
```

Comment:
Could we leave this code in the same file as the `AutoTLSPlugin` file?

Reply:
The only reason that I broke it out into anther file was due it being copy/pasted from a source with a different license than ours. Suggestions on how to properly attribute/license this copy/pasted code if it were in the same file? Same object(`AutoTLSCross`) in `AutoTLSPlugin` file with the attribution inside the object[is that enough/acceptable?]?

Reply:
I'd just add a comment above the copy/pasted code that links to the original source, indicates the original license, and then mentions that it's only copied here because there was no other easy way to achieve the same effect.